src/coreclr/src/vm/ecall.cpp

```diff
@@ -195,6 +195,25 @@ void ECall::PopulateManagedCastHelpers()
     pDest = pMD->GetMultiCallableAddrOfCode();
     SetJitHelperFunction(CORINFO_HELP_UNBOX, pDest);
 
+    // Array element accessors are more perf sensitive than other managed helpers and indirection 
+    // costs introduced by PreStub could be noticeable (7% to 30% depending on platform).
```

Comment:
Add a comment that this should be revisited once https://github.com/dotnet/runtime/issues/5857 is fixed?